GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace GenericsHomework;
+public class Node<T>
+{
+    private int Size { get; set; }
+    public T? Item { get; set; }
+    public Node<T> Next
+    {get; private set; }
+
+    public Node(T? item, Node<T> next)
+    {
+        Item = item;
+        Next = next ?? this;
```

Comment:
```suggestion
        Next = this;
```
In the constructor, next should just be set to `this`